### PR TITLE
Add authorization rules

### PR DESCRIPTION
**Pull request by mryeminthu**
- Install CanCanCan.
- Add a role column to the users table. 
- A user can delete a post if it is theirs or if they have an admin role.
-  Add the "Delete" button to the view.
- A user can delete a comment if it is theirs or if they have an admin role.

The reason for being a solo project.
Untitled

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -27,6 +27,19 @@ def show
     @user = @post.author
   end
 
+  def destroy
+    @user = User.find(params[:user_id])
+    @post = @user.posts.find_by(id: params[:id])
```
Comment (thecodechaser) [inline app/controllers/posts_controller.rb:32]: - [ ] Please make sure the app uses `CanCanCan` for Post authorization in Post Controller. 
  - You can consider using the below code:
  
```ruby
 class PostsController < ApplicationController
    load_and_authorize_resource
    ...
  end
```

File: app/models/ability.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    user ||= User.new # guest user (not logged in)
+
+    if user.admin?
+      can :manage, :all
```
Comment (thecodechaser) [inline app/models/ability.rb:8]: - [ ] You are calling the method `admin?` from the user model but this method is not present there. Please make sure to create this method in the user model.
  - You can consider using the below code:
  
```ruby
 def admin?
     role == 'admin' ? true : false
   end
```

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @mryeminthu👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- Good job with authorization. ✔️
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Ibizugbe — APPROVED** [review]
Hi @mryeminthu 👋🏼👋🏼👋🏼, 

approved


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights✅✅✅

- [x] App has been updated to make use of `can can can`👍🏼.
- [x] All suggestions from the previous reviewer have been implemented👍🏼.
- [x] App works well with admin roles👍🏼.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**